pkg/s3wrapper/client.go

```diff
@@ -33,23 +33,28 @@ const (
 	awsEndpointSuffix = ".amazonaws.com"
 )
 
+type ObjectInfo struct {
+	Path     string
+	Metadata map[string]*string
+}
+
 //go:generate mockgen --build_flags=--mod=mod -package=s3wrapper -destination=mock_s3wrapper.go . API
 //go:generate mockgen --build_flags=--mod=mod -package s3wrapper -destination mock_s3iface.go github.com/aws/aws-sdk-go/service/s3/s3iface S3API
 //go:generate mockgen --build_flags=--mod=mod -package s3wrapper -destination mock_s3manageriface.go github.com/aws/aws-sdk-go/service/s3/s3manager/s3manageriface UploaderAPI
 type API interface {
 	IsAwsS3() bool
 	CreateBucket() error
-	Upload(ctx context.Context, data []byte, objectName string) error
-	UploadStream(ctx context.Context, reader io.Reader, objectName string) error
-	UploadFile(ctx context.Context, filePath, objectName string) error
+	Upload(ctx context.Context, data []byte, objectName string, metadata map[string]*string) error
```

Comment:
Is the intent here to allow `nil` values? What would be the meaning of that? May be worth a documentation comment.

Reply:
This is how metadata is defined in the S3 API, I don't believe the intent is to allow nil values.

Reply:
See previous comment

Reply:
Closing as the whole topic of map[string]*string vs map[string]string is dealt with in another comment.